dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/utils/ReferenceConfigCache.java

```diff
@@ -175,7 +175,8 @@ public void destroy(String key, Class<?> type) {
 
         Map<String, Object> proxiesOftype = proxies.get(type);
         if (CollectionUtils.isNotEmptyMap(proxiesOftype)) {
-            proxiesOftype.remove(key);
+            Destroyable proxy = (Destroyable) proxiesOftype.remove(key);
```

Comment:
it would be better to check if the object is an instance of Destroyable first

Reply:
The generated proxy instances all automatically implement the Destroyable interface, right?

![image](https://user-images.githubusercontent.com/43363120/122699402-ee8d9a80-d27b-11eb-9d50-7bf726eb635f.png)
![image](https://user-images.githubusercontent.com/43363120/122699411-f2b9b800-d27b-11eb-9e6c-0158dd33896e.png)